server/services/WalletService.js

```diff
@@ -1,24 +1,41 @@
-const WalletRepository = require("../repositories/WalletRepository");
-const Wallet = require("../models/Wallet");
-const expect = require("expect-runtime");
+const WalletRepository = require('../repositories/WalletRepository');
+const Wallet = require('../models/Wallet');
+const HttpError = require("../utils/HttpError");
+const expect = require('expect-runtime');
 
-class WalletService{
-  constructor(){
+class WalletService {
+  constructor() {
     this.walletRepository = new WalletRepository();
   }
 
-  async getById(id){
+  async getById(id) {
     const object = await this.walletRepository.getById(id);
+    expect(object).match({ id: expect.any(Number) });
     const wallet = new Wallet(object.id);
     return wallet;
   }
 
-  async getByName(name){
+  async getByName(name) {
     const object = await this.walletRepository.getByName(name);
-    expect(object).match({id:expect.any(Number)});
+    expect(object).match({ id: expect.any(Number) });
     const wallet = new Wallet(object.id);
     return wallet;
   }
+
+  async getByIdOrName(idOrName) {
```

Comment:
Added this method in WalletServices to handle the cases where we are not sure whether we have a number or string. 
We are also now using typeof instead of `isNaN`